### PR TITLE
HIVE-28096: Fix the mismatched function names in HMSHandler audit log

### PR DESCRIPTION
**Pull request by wecharyu**
### What changes were proposed in this pull request?
Normalize the `startFunction()` and `endFunction()` calls in `HMSHandler`:
1. Fix the mismatched function name.
2. Add the missed `startFunction()` or endFunction() calls.
3. Unify the function name, remove the `.(dot)` or additional suffix empty chars in some function names.

### Why are the changes needed?
The mismatched function names would cause the counter and api cost metrics wrong.

### Does this PR introduce _any_ user-facing change?
No.

### Is the change a dependency upgrade?
No.

### How was this patch tested?
Deploy and test the metrics.

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ public GetCatalogsResponse get_catalogs() throws MetaException {
       ex = e;
       throw e;
     } finally {
-      endFunction("get_catalog", ret != null, ex);
+      endFunction("get_catalogs", ret != null, ex);
```
Comment (zratkai) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java:1061]: This should be "alter_catalog" .

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java [hunk 2]
```diff
@@ -4392,19 +4392,17 @@ private List<Partition> createPartitionFolders(final List<Partition> partitionsT
...
     String catName = request.isSetCatName() ? request.getCatName() : getDefaultCatalog(conf);
     String dbName = request.getDbName();
     String tblName = request.getTblName();
     startTableFunction("add_partitions_req", catName, dbName, tblName);
 
+    AddPartitionsResult result = new AddPartitionsResult();
     Exception ex = null;
     try {
+      if (request.getParts().isEmpty()) {
+        return result;
```
Comment (zratkai) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java:4404]: This is not covered in the topic of thic PR: " Fix the mismatched function names".
Why do we need this here?

**Review by zratkai — COMMENTED** [review]
Please review my finding.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java (hunk 2 above)
Comment (wecharyu) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java:4404]: If we do not move this code into try clause, the `endFunction()` calls would be missed.

**Pull request closed without merging**